clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -10904,14 +10911,18 @@ bool Sema::CheckDestructor(CXXDestructorDecl *Destructor) {
       // conversion from 'this' to the type of a destroying operator delete's
       // first parameter, perform that conversion now.
       if (OperatorDelete->isDestroyingOperatorDelete()) {
-        QualType ParamType = OperatorDelete->getParamDecl(0)->getType();
+        unsigned PointerParam = 0;
```

Comment:
```suggestion
        unsigned AddressParamIndex = 0;
```
Again, the "get the index of the address parameter" could be a small function